src/CoreBundle/private/library/classes/TCMSListManager/TCMSListManagerFullGroupTable.class.php

```diff
@@ -99,21 +89,48 @@ public function Init(&$oTableConf)
             $this->PostCreateTableObjectHook();
         }
 
-        if (($isTableCacheChangeRequest && null === $oOldTableObj) && $this->bListCacheEnabled && CMS_ACTIVE_BACKEND_LIST_CACHE) {
-            $tmp = serialize($this->tableObj);
-            $tmp = gzcompress($tmp, 9);
-            $_SESSION['_listObjCache'][$sListCacheKey] = base64_encode($tmp);
+        if (true === $this->isTableCachingEnabled() && false === $this->isModuleFunction() && (true === $this->isTableCacheChangeRequest() || null === $cachedTableObj)) {
+            $this->saveTableInSessionCache($listCacheKey, $this->tableObj);
         }
     }
 
-    private function isTableCacheChangeRequest(): bool
+    private function isTableCachingEnabled(): bool
     {
-        $inputFilter = $this->getInputFilterUtil();
+        return $this->bListCacheEnabled && CMS_ACTIVE_BACKEND_LIST_CACHE;
+    }
 
-        if (null !== $inputFilter->getFilteredInput('_fnc')) {
-            return false;
+    private function getTableFromSessionCache(string $cacheKey): ?TFullGroupTable
+    {
+        if (false === \array_key_exists('_listObjCache', $_SESSION)) {
+            $_SESSION['_listObjCache'] = [];
```

Comment:
if the key does not exists, then we can early exit, can we not?

Reply:
True. Changed.

(It was a replica of the old code which also had exactly this case.)